### DISCUSSION TITLE
Editcontext IsModified & HxSearchBox

### DISCUSSION THREAD

**Discussion by nironixon** [havit/Havit.Blazor#420] [Q&A]
Hi there, 
I having a slight issue with the HxSearchBox component and when we are using the Editcontext.
Despite not changing any property on the model that is bound to the Editcontext the form is marked as modified whenever i 
use the HxSearchBox. Shouldn't it be possible to use the HxSearchBox without marking the context as modified?

Below i have put togheter an example that illustrates my problem, whenever i touch the HxSearchBox the submit button is enabled. 

Best Regards
Nic

`
@page "/validation"

<EditForm EditContext="editContext" OnValidSubmit="@HandleValidSubmit">
    <DataAnnotationsValidator />
    <ValidationSummary />

    <HxSearchBox DataProvider="ProvideSearchResults"
                 TItem="SearchBoxItem"
                 ItemTitleSelector="i => i.Title"
                 ItemSubtitleSelector="i => i.Subtitle"
                 ItemIconSelector="i => i.Icon"
                 Placeholder="Search"
                 OnItemSelected="OnItemSelected"
                 OnTextQueryTriggered="OnTextQueryTriggered">
        <DefaultContentTemplate>
            <div class="p-2 text-muted">Search for Mouse, Table or Door...</div>
        </DefaultContentTemplate>
        <NotFoundTemplate>
            <div class="small py-2 px-3 text-muted"><HxIcon CssClass="me-2" Icon="BootstrapIcon.EmojiFrown" />Sorry, I did not find that...</div>
        </NotFoundTemplate>
    </HxSearchBox>

    <p class="mt-3">
        Last selected item: @selectedItem?.Title<br />
        Last triggered text-query: @triggeredTextQuery<br />
    </p>

    <HxInputText Label="Email" @bind-Value="@model.Name" InputSize="InputSize.Small" CssClass="mb-3" />
    <button class="btn btn-sm btn-bop-primary" type="submit" Disabled="@(!context.IsModified())">Spara </button>
</EditForm>

@code {
    private EditContext? editContext;
    private SearchBoxItem selectedItem;
    private string triggeredTextQuery;
    private Person? model;

    List<SearchBoxItem> Data { get; set; } = new()
    {
        new() { Title = "Table", Subtitle = "50 000", Icon = BootstrapIcon.Table },
        new() { Title = "Mouse", Subtitle = "400", Icon = BootstrapIcon.Mouse },
        new() { Title = "Door", Subtitle = "1000", Icon = BootstrapIcon.DoorClosed },
        new() { Title = "Big table", Subtitle = "9 000", Icon = BootstrapIcon.Table },
        new() { Title = "Small table", Subtitle = "7 200", Icon = BootstrapIcon.Table }
    };

    private void OnItemSelected(SearchBoxItem item)
    {
        selectedItem = item;
    }

    private void OnTextQueryTriggered(string text)
    {
        triggeredTextQuery = text;
    }

    private async Task<SearchBoxDataProviderResult<SearchBoxItem>> ProvideSearchResults(SearchBoxDataProviderRequest request)
    {
        return new()
            {
                Data = Data.Where(i => i.Title.Contains(request.UserInput, StringComparison.OrdinalIgnoreCase))
            };
    }

    internal class SearchBoxItem
    {
        public string Title { get; set; }
        public string Subtitle { get; set; }
        public BootstrapIcon Icon { get; set; }
    }

    internal class Person
    {
        public string Name { get; set; }
    }

    protected override async Task OnInitializedAsync()
    {
        this.model = new Person();
        this.editContext = new EditContext(this.model);
        this.editContext.OnFieldChanged += this.HandleFieldChanged;
    }

    void HandleFieldChanged(object? sender, FieldChangedEventArgs e)
    {
        editContext.MarkAsUnmodified();
        if (this.editContext is not null)
        {
          //  this.formValidForSubmission = this.editContext.Validate();
           // this.StateHasChanged();
        }
    }

    async Task HandleValidSubmit()
    {

    }
}

`

**Comment by hakenr** [comment]
Hi @nironixon,
can you please provide more complete sample to be able to reproduce do the issue? The code provided above does not use the `editContext` field at al. There is some call from the code of the `<button Disabled="context.IsModified()" ... />`, but the `context` there is not used anywhere near (especially not for the `HxSearchBox`).

The `HxSearchBox` should not influence the `EditContext` as it does not reference/use it in any internal part.
